### PR TITLE
CFE-68 : Update EC2 instace with Infracture resource tags, consider s…

### PR DESCRIPTION
**Pull request by Connect2naga**
issue :
Currently the user-defined tags during install are passed directly as parameters of the Machine and Machineset resources for the master and worker. As a result these tags cannot be updated by consulting the Infrastructure resource of the cluster where the user defined tags are written.

so intention of the changes is to update EC2 instance,
Solution :
The machine-api-provider-aws update the EC2 instance tags from infrastructure resource on periodical bases. when the resource tags are updated it should update the tags on the EC2 instances without restarts.

Part 1 : Fetch tags from resource tags under infrastructure status. [Done, merged.]
Part 2 : Fetch tags from resource tags under infrastructure status and Spec, Then apply on merged tags. [Current.]

target Release : 4.11

### CODE REVIEW

**Comment by alexander-demichev** [comment]
I'll put it on hold before we merge API PR.
/hold

File: pkg/actuators/machine/utils.go [hunk 1]
```diff
@@ -486,11 +486,22 @@ func ProviderStatusFromRawExtension(rawExtension *runtime.RawExtension) (*machin
...
+		infra.Status.PlatformStatus.AWS != nil {
+		for _, value := range infra.Status.PlatformStatus.AWS.ResourceTags {
+			mergedTags[value.Key] = value.Value
+		}
+	}
+
+	if infra != nil && infra.Spec.PlatformSpec.AWS != nil {
+		for _, value := range infra.Spec.PlatformSpec.AWS.ResourceTags {
+			mergedTags[value.Key] = value.Value
+		}
```
Comment (JoelSpeed) [inline pkg/actuators/machine/utils.go:503]: Why are we reading from the Spec and the Status? I was under the impression some other controller would sync tags on the Spec to the Status and that we would just read from the Status?

**Review by JoelSpeed — COMMENTED** [review]
Is this PR actually needed? We are reading from the status already, based on the proposal I thought we would continue to just read from the Status?

**Pull request closed without merging**